packages/node/src/plugins/autostaker/AutostakerPlugin.ts

```diff
@@ -119,15 +120,49 @@ export class AutostakerPlugin extends Plugin<AutostakerPluginConfig> {
         }
         logger.info(`First activation in approximately ${this.pluginConfig.runIntervalInMs / (1000 * 60) } minutes`)
         scheduleAtApproximateInterval(triggerRun, this.pluginConfig.runIntervalInMs, 0.1, false, this.abortController.signal)
-        streamrClient.on('sponsorshipCreated', triggerRun)
+        streamrClient.on('sponsorshipCreated', (event: SponsorshipCreatedEvent) => {
+            // Make sure the The Graph is up-to-date before triggering the run
+            logger.info('Detected a new sponsorship at block number', { blockNumber: event.blockNumber })
+            streamrClient.getTheGraphClient().updateRequiredBlockNumber(event.blockNumber)
+            triggerRun()
+        })
         this.abortController.signal.addEventListener('abort', () => {
             streamrClient.off('sponsorshipCreated', triggerRun)
         })
     }
 
+    // Broadcasts a transaction corresponding to the action without waiting for it to be mined
+    private async submitAction(action: Action, signer: SignerWithProvider): Promise<ContractTransactionResponse> {
+        logger.info(`Execute action: ${action.type} ${formatEther(action.amount)} ${action.sponsorshipId}`)
+        const stakeOrUnstakeFunction = getStakeOrUnstakeFunction(action)
+        return new Promise((resolve, reject) => {
+            stakeOrUnstakeFunction(signer,
+                this.pluginConfig.operatorContractAddress,
+                action.sponsorshipId,
+                action.amount,
+                (tx) => resolve(tx) // resolve onSubmit
+            ).catch(reject)
+        })
+    }
+
+    // This will retry the transaction preflight checks, defends against various transient errors
+    private async submitActionWithRetry(action: Action, signer: SignerWithProvider, 
```

Comment:
`maxRetries` and `retryDelay` are never used by caller, so we could just use values of 5 and 5000 at lines 157-158. Preferably via constants (e.g. `ACTION_SUBMIT_RETRY_COUNT` and   `ACTION_SUBMIT_RETRY_DELAY`)